Source: oahong@oahong.me
Upstream: no, this is not a proper fix
Reason: Use system enet
--- supertuxkart-0.7/configure.ac.orig	2011-02-03 14:29:07.445472650 +0800
+++ supertuxkart-0.7/configure.ac	2011-02-03 15:29:44.901473040 +0800
@@ -399,7 +399,6 @@
   ;;
 esac
 SUMMARY="$SUMMARY\nUsing enet network support."
-ENETTREE="src/enet"
 
 # =================
 # Check for gettext
@@ -437,7 +436,6 @@
 AC_SUBST(openal_LIBS)
 AC_SUBST(oggvorbis_LIBS)
 AC_SUBST(BULLETTREE)
-AC_SUBST(ENETTREE)
 
 # ================
 # Create makefiles
@@ -459,8 +457,7 @@
  src/Makefile                       \
  src/ide/Makefile                   \
  src/bullet/Makefile                \
- src/bullet/src/Makefile            \
- src/enet/Makefile                  \
+ src/bullet/src/Makefile
 ])
 AC_OUTPUT
 echo -e $SUMMARY
--- supertuxkart-0.7/Makefile.am.orig	2011-02-03 15:29:52.331473020 +0800
+++ supertuxkart-0.7/Makefile.am	2011-02-03 15:30:09.143473002 +0800
@@ -1,6 +1,6 @@
 ACLOCAL_AMFLAGS = -I m4
 
-SUBDIRS = $(BULLETTREE) $(ENETTREE) src doc data
+SUBDIRS = $(BULLETTREE) src doc data
 
 pkgdatadir=$(datadir)/doc/$(PACKAGE)
 dist_pkgdata_DATA=AUTHORS ChangeLog COPYING README
--- supertuxkart-0.7/src/network/network_manager.cpp.orig	2011-02-03 18:04:29.666473002 +0800
+++ supertuxkart-0.7/src/network/network_manager.cpp	2011-02-03 18:08:32.277473002 +0800
@@ -82,6 +82,7 @@
 
      m_host = enet_host_create (& address     /* the address to bind the server host to */, 
                                 stk_config->m_max_karts /* number of connections */,
+				0             /* channels */,
                                 0             /* incoming bandwidth */,
                                 0             /* outgoing bandwidth */     );
     if (m_host == NULL)
@@ -107,6 +108,7 @@
 {
     m_host = enet_host_create (NULL /* create a client host */,
                                1    /* only allow 1 outgoing connection */,
+			       0    /* chanels */,
                                0    /* downstream bandwidth unlimited   */,
                                0    /*  upstream bandwidth unlimited    */ );
     
@@ -125,7 +127,7 @@
     address.port = UserConfigParams::m_server_port;
 
     /* Initiate the connection, allocating the two channels 0 and 1. */
-    peer = enet_host_connect (m_host, &address, 2);    
+    peer = enet_host_connect (m_host, &address, 2, NULL);    
     
     if (peer == NULL)
     {
--- ../../supertuxkart-0.7/src/Makefile.am.orig	2011-02-03 19:59:36.686473001 +0800
+++ ../../supertuxkart-0.7/src/Makefile.am	2011-02-03 20:04:01.317472999 +0800
@@ -2,7 +2,6 @@
 
 SUBDIRS = ide
 
-bindir=$(prefix)/games
 bin_PROGRAMS = supertuxkart
 
 AM_CPPFLAGS = -DSUPERTUXKART_DATADIR="\"$(datadir)/games/$(PACKAGE)/\""  \
